### PR TITLE
Allow custom loggers in Driver, use MSBuild log in ILLink

### PR DESCRIPTION
**Pull request by JosephTremoulet**
Currently, attempts to log messages when the linker is invoked by the ILLink MSBuild task get swallowed because MSBuild swallows tasks' stdout.  The linker already has an ILogger interface, so enable passing one to the Driver, and update the ILLink task to pass one to the Driver which will forward messages to any attached MSBuild loggers.

### CODE REVIEW

**Comment by JosephTremoulet** [comment]
@sven @marek-safar PTAL

**Comment by sbomer** [comment]
LGTM

File: linker/Linker/ILogger.cs [hunk 1]
```diff
@@ -12,4 +12,19 @@ public interface ILogger
 	{
 		void LogMessage (MessageImportance importance, string message, params object[] values);
 	}
+
+	public class LambdaLogger : ILogger
```
Comment (marek-safar) [inline linker/Linker/ILogger.cs]: This does not make much sense to me. Why do have a double redirect? Please just implement ILogger in ILLinker the way you want

File: linker/Linker/Driver.cs [hunk 2]
```diff
@@ -75,10 +80,14 @@ bool HaveMoreTokens ()
 			return _queue.Count > 0;
 		}
 
-		public void Run ()
+		public void Run (ILogger customLogger = null)
 		{
 			Pipeline p = GetStandardPipeline ();
 			using (LinkContext context = GetDefaultContext (p)) {
+				if (customLogger != null)
```
Comment (marek-safar) [inline linker/Linker/Driver.cs]: nit: follow the coding style

**Comment by JosephTremoulet** [comment]
@marek-safar thanks for the review, updated per your suggestions (I think -- if "follow the coding style" meant something other than "remove the braces around the single-line `if` body", please LMK)

**Pull request merged**